spark/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -68,15 +68,17 @@ private[delta] case class CurrentTransactionInfo(
    */
   lazy val finalActionsToCommit: Seq[Action] = commitInfo ++: actions
 
-  var newMetadata: Option[Metadata] = None
-  var newProtocol: Option[Protocol] = None
```

Comment:
Why remove this/why was this needed in the first place?

Reply:
previously we were passing only newMetadata and newProtocol when they used to change. Now we pass metadata and protocol (old and new) to commit Store for more flexibility.